### PR TITLE
llvm: convert more things to use Builder

### PR DESCRIPTION
**Pull request by jacobly0**
* finish converting intrinsics
 * finish converting attributes
 * finish converting instructions
 * finish converting globals
 * pass behavior tests with no dependence on the llvm api (`-fno-libllvm`)

### CODE REVIEW

File: src/codegen/llvm/Builder.zig [hunk 1]
```diff
@@ -6349,8 +7038,11 @@ pub fn init(options: Options) InitError!Builder {
         inline for (.{ 1, 8, 16, 29, 32, 64, 80, 128 }) |bits|
             assert(self.intTypeAssumeCapacity(bits) ==
                 @field(Type, std.fmt.comptimePrint("i{d}", .{bits})));
-        inline for (.{0}) |addr_space|
-            assert(self.ptrTypeAssumeCapacity(@enumFromInt(addr_space)) == .ptr);
+        inline for (.{ 0, 4 }) |addr_space_index| {
```
Comment (andrewrk) [inline src/codegen/llvm/Builder.zig:7041]: do these need to be `inline`? doesn't look like you're depending on the comptime-ness to me.

**Review by andrewrk — APPROVED** [review]
Nice to see all those API calls disappear.

File: src/codegen/llvm/Builder.zig (hunk 1 above)
Comment (jacobly0) [reply src/codegen/llvm/Builder.zig:7127]: Without the asserts, `inline` would not be needed.  What I really want is to transform the array at comptime and then iterate the transformed array at runtime.

File: src/codegen/llvm/Builder.zig [hunk 2]
```diff
@@ -2279,6 +2305,906 @@ pub const Variable = struct {
...
+                .{ .kind = .{ .matches = 0 } },
+            },
+            .attrs = &.{ .nocallback, .nofree, .nosync, .nounwind, .speculatable, .willreturn, .{ .memory = Attribute.Memory.all(.none) } },
+        },
+        .ctlz = .{
+            .ret_len = 1,
+            .params = &.{
+                .{ .kind = .overloaded },
+                .{ .kind = .{ .matches = 0 } },
+                .{ .kind = .{ .type = .i1 } },
```
Comment (antlilja) [inline src/codegen/llvm/Builder.zig:2886]: Shouldn't this ```is_zero_poison``` argument and the same one for ```cttz``` have the ```immarg``` attribute like the ```abs``` intrinsic?

The langref says it has to be a ["constant flag"](https://llvm.org/docs/LangRef.html#id654). The ```abs``` intrinsic has a similar but different wording, the argument has to be a ["constant and a flag"](https://llvm.org/docs/LangRef.html#id488).

**Pull request merged**